plugins/devtools/src/components/Content/TasksContent/TasksContent.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Alert } from '@material-ui/lab';
+import React from 'react';
+import { Progress } from '@backstage/core-components';
+import { useTasks } from '../../../hooks/useTasks';
+import { Table, TableColumn } from '@backstage/core-components';
+import { TaskInfo } from '@backstage/plugin-devtools-common';
+import { IconButton, Tooltip, Typography } from '@material-ui/core';
+import { devToolsApiRef } from '../../../api';
+import { alertApiRef, useApi } from '@backstage/core-plugin-api';
+import RefreshIcon from '@material-ui/icons/Refresh';
+
+const TriggerButton = (props: { task: TaskInfo }) => {
+  const { task } = props;
+  const api = useApi(devToolsApiRef);
+  const alertApi = useApi(alertApiRef);
+  const handleClick = () => {
+    api
+      .triggerTask(task.scheduler, task.id)
```

Comment:
I know you have this secured on the backend API Route, but would it make sense to also maybe show or hide this based on the permission as well? Mostly makes more sense to do if you add an update permission though